### PR TITLE
User service integrate with API

### LINKED ISSUE #23: Integrate user service with API

### PR DESCRIPTION
**Pull request by Duveshka**

### CODE REVIEW

File: ArcadiaParties-SPA/src/app/user/services/user.service.ts [hunk 1]
```diff
@@ -1,23 +1,22 @@
 import { Injectable } from '@angular/core';
 import { User } from '../models/User';
+import { environment } from 'src/environments/environment';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
```
Comment (laska-sl) [inline ArcadiaParties-SPA/src/app/user/services/user.service.ts]: Internal after external

File: ArcadiaParties-SPA/src/app/user/effects/effect.ts [hunk 2]
```diff
@@ -11,7 +11,10 @@ export class UserEffect {
   loadUser = createEffect(() =>
```
Comment (laska-sl) [inline ArcadiaParties-SPA/src/app/user/effects/effect.ts:10]: 7-8 empty line. Can't highlight, cause they were here before this PR

File: ArcadiaParties-SPA/src/app/user/effects/effect.ts [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
 
-
 @Injectable()
 export class UserEffect {
   loadUser = createEffect(() =>
     this.actions.pipe(
       ofType(loadUserAction),
-      map(() => loadUserSuccessAction({user : this.userService.getUser()}))
+      mergeMap(() => this.userService.getUser()
+      .pipe(
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/user/effects/effect.ts]: You don't need to use pipe again.

File: ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs [hunk 4]
```diff
@@ -10,6 +10,6 @@ public interface IUserRepository
 
         Task<IEnumerable<UserDTO>> GetUsers();
 
-        Task<IEnumerable<UserDTO>> GetUsersOfDepartment(int departmentId);
+        Task<IEnumerable<UserForCalendarDTO>> GetUsersOfDepartment(int departmentId);
```
Comment (sponkrashin) [inline ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs]: User of departments aren't related to calendar, they are still just users of the department. It doesn't make sense.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/src/app/user/effects/effect.ts (hunk 2 above)
Comment (Duveshka) [reply ArcadiaParties-SPA/src/app/user/effects/effect.ts:10]: It deleted

File: ArcadiaParties-SPA/src/app/user/services/user.service.ts (hunk 1 above)
Comment (Duveshka) [reply ArcadiaParties-SPA/src/app/user/services/user.service.ts]: Fixed

File: ArcadiaParties-SPA/src/app/user/effects/effect.ts (hunk 3 above)
Comment (Duveshka) [reply ArcadiaParties-SPA/src/app/user/effects/effect.ts]: Fixed

File: ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs (hunk 4 above)
Comment (Duveshka) [reply ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs]: Renamed

File: ArcadiaParties.Data.Abstractions/DTOs/UsersOfDepartmentDTO.cs [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 namespace ArcadiaParties.Data.Abstractions.DTOs
 {
-    public class UserForCalendarDTO
```
Comment (sponkrashin) [inline ArcadiaParties.Data.Abstractions/DTOs/UsersOfDepartmentDTO.cs]: Revert it back.

File: ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs [hunk 6]
```diff
@@ -10,6 +10,6 @@ public interface IUserRepository
 
         Task<IEnumerable<UserDTO>> GetUsers();
 
-        Task<IEnumerable<UserDTO>> GetUsersOfDepartment(int departmentId);
+        Task<IEnumerable<UsersOfDepartmentDTO>> GetUsersOfDepartment(int departmentId);
```
Comment (sponkrashin) [inline ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs]: Revert it back.

File: ArcadiaParties.API/Controllers/CalendarController.cs [hunk 7]
```diff
@@ -21,7 +21,7 @@ public CalendarController(IMediator mediator)
             _mediator = mediator;
         }
 
-        [ProducesResponseType(typeof(IEnumerable<UserForCalendarDTO>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(IEnumerable<UsersOfDepartmentDTO>), StatusCodes.Status200OK)]
```
Comment (sponkrashin) [inline ArcadiaParties.API/Controllers/CalendarController.cs]: Revert it back.

File: ArcadiaParties.Data/Repositories/UserRepository.cs [hunk 8]
```diff
@@ -45,10 +45,10 @@ public async Task<IEnumerable<UserDTO>> GetUsers()
             return usersToReturn;
         }
 
-        public async Task<IEnumerable<UserDTO>> GetUsersOfDepartment(int departmentId)
+        public async Task<IEnumerable<UsersOfDepartmentDTO>> GetUsersOfDepartment(int departmentId)
```
Comment (sponkrashin) [inline ArcadiaParties.Data/Repositories/UserRepository.cs]: Revert it back.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties.Data/Repositories/UserRepository.cs (hunk 8 above)
Comment (Duveshka) [reply ArcadiaParties.Data/Repositories/UserRepository.cs]: Reverted

File: ArcadiaParties.API/Controllers/CalendarController.cs (hunk 7 above)
Comment (Duveshka) [reply ArcadiaParties.API/Controllers/CalendarController.cs]: Reverted

File: ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs (hunk 6 above)
Comment (Duveshka) [reply ArcadiaParties.Data.Abstractions/Repositories/IUserRepository.cs]: Reverted

File: ArcadiaParties.Data.Abstractions/DTOs/UsersOfDepartmentDTO.cs (hunk 5 above)
Comment (Duveshka) [reply ArcadiaParties.Data.Abstractions/DTOs/UsersOfDepartmentDTO.cs]: Reverted

File: ArcadiaParties.Data.Abstractions/DTOs/UsersForCalendarDTO.cs [hunk 9]
```diff
@@ -3,7 +3,7 @@
 
 namespace ArcadiaParties.Data.Abstractions.DTOs
 {
-    public class UserForCalendarDTO
```
Comment (sponkrashin) [inline ArcadiaParties.Data.Abstractions/DTOs/UsersForCalendarDTO.cs]: Revert naming.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties.Data.Abstractions/DTOs/UsersForCalendarDTO.cs (hunk 9 above)
Comment (Duveshka) [reply ArcadiaParties.Data.Abstractions/DTOs/UsersForCalendarDTO.cs]: Reverted

**Review by sponkrashin — DISMISSED** [review]
Wait until base PR is merged. Then merge this PR to develop.

**Review by sponkrashin — APPROVED** [review]

**Pull request merged**